pkg/gardenlet/controller/backupbucket/backup_bucket.go

```diff
@@ -56,45 +60,40 @@ type Controller struct {
 // NewBackupBucketController takes a Kubernetes client for the Garden clusters <k8sGardenClient>, a struct
 // holding information about the acting Gardener, a <backupBucketInformer>, and a <recorder> for
 // event recording. It creates a new Gardener controller.
-func NewBackupBucketController(clientMap clientmap.ClientMap, k8sGardenCoreInformers gardencoreinformers.SharedInformerFactory, config *config.GardenletConfiguration, recorder record.EventRecorder) *Controller {
-	var (
-		gardencorev1beta1Informer = k8sGardenCoreInformers.Core().V1beta1()
-		backupBucketInformer      = gardencorev1beta1Informer.BackupBuckets()
-
-		seedInformer = gardencorev1beta1Informer.Seeds()
-		seedLister   = seedInformer.Lister()
-	)
-
-	backupBucketController := &Controller{
-		config:            config,
-		reconciler:        newReconciler(context.TODO(), clientMap, recorder, config),
-		recorder:          recorder,
-		backupBucketQueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "BackupBucket"),
-		workerCh:          make(chan int),
+func NewBackupBucketController(ctx context.Context, clientMap clientmap.ClientMap, config *config.GardenletConfiguration, recorder record.EventRecorder) (*Controller, error) {
+	gardenClient, err := clientMap.GetClient(ctx, keys.ForGarden())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get garden client: %w", err)
 	}
 
-	backupBucketInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: controllerutils.BackupBucketFilterFunc(confighelper.SeedNameFromSeedConfig(config.SeedConfig), seedLister, config.SeedSelector),
-		Handler: cache.ResourceEventHandlerFuncs{
-			AddFunc:    backupBucketController.backupBucketAdd,
-			UpdateFunc: backupBucketController.backupBucketUpdate,
-			DeleteFunc: backupBucketController.backupBucketDelete,
-		},
-	})
-
-	backupBucketController.backupBucketSynced = backupBucketInformer.Informer().HasSynced
```

Comment:
Can you revert the removal of the hasSycned handling? IIUC, we agreed on checking for the synced state since it is part of the informer interface and the fact that we don't need it at the moment, because the Controller-Runtime never returns an unsynced informer, is only an implementation detail.

Reply:
Yes, we agreed on that one. I just forgot to adapt it here.
Done now :)